src/redux/slices/ensResolver.slice.ts

```diff
@@ -0,0 +1,114 @@
+import urlcat from 'urlcat'
+import { ethers } from "ethers";
+import { createApi,  fakeBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export interface NameInfo {
+  rnsName: string
+  ensName: string | null
+  address: string
+}
+
+export interface ProfileInfo {
+  avatar: string[]
+  bio: string
+  name: string
+}
+
+interface RSS3Info {
+  profile: ProfileInfo
+}
+
+
+export const getNameById = async (id: string ) => {
+  if (!id) return ''
+    const url = urlcat('https://rss3.domains/address/:id', { id })
```

Comment:
I didn't know about RSS3. It's interesting... but I'd lean on the side of not using it at all. It would essentially be a niche centralized service we'd be relying on. For querying ENS data in not large bulks, Ethers.js along with mainnet RPC is good enough...